server/store/datastore/registry.go

```diff
@@ -26,9 +26,9 @@ func (s storage) RegistryFind(repo *model.Repo, addr string) (*model.Registry, e
 	return reg, wrapGet(s.engine.Get(reg))
 }
 
-func (s storage) RegistryList(repo *model.Repo) ([]*model.Registry, error) {
-	regs := make([]*model.Registry, 0, perPage)
-	return regs, s.engine.Where("registry_repo_id = ?", repo.ID).Find(&regs)
+func (s storage) RegistryList(repo *model.Repo, p *model.ListOptions) ([]*model.Registry, error) {
+	var regs []*model.Registry
+	return regs, s.paginate(p).Where("registry_repo_id = ?", repo.ID).Find(&regs)
```

Comment:
It's unrelated ...  I would vote for an extra pull/issue!

Reply:
Partly, when there is no paging it does not matter so much if order is not stable but when there is paging it can be that when opening second page you get results again from first page